run_cart.sh

```diff
@@ -101,47 +104,43 @@ prepare(){
-    cat Log/$SLURM_JOB_ID/slurm_hostlist | head -$DAOS_CLIENTS > Log/$SLURM_JOB_ID/daos_client_hostlist
+    ./gen_hostlist.sh $N_SERVERS $N_CLIENTS
 }
 
 run_selftest(){
-    last_srv_index=$(( DAOS_SERVERS-1 ))
-    run_bg $DAOS_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "${CARTDIR}/install/Linux/bin/crt_launch -e ${CARTDIR}/install/Linux/TESTING/tests/test_group_np_srv --name selftest_srv_grp_$DAOS_SERVERS --cfg_path=${CARTDIR}/install/Linux/TESTING" $test
+    last_srv_index=$(( $N_SERVERS-1 ))
+    run_bg $N_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "${CARTDIR}/install/Linux/bin/crt_launch -e ${CARTDIR}/install/Linux/TESTING/tests/test_group_np_srv --name selftest_srv_grp_$N_SERVERS --cfg_path=${CARTDIR}/install/Linux/TESTING" $test
     wait 5
-    run_st $DAOS_CLIENTS Log/$SLURM_JOB_ID/daos_client_hostlist "${CARTDIR}/install/Linux/bin/self_test --group-name selftest_srv_grp_$DAOS_SERVERS --endpoint 0-$last_srv_index:0 --master-endpoint 0:0 --max-inflight-rpcs $MAX_INFLIGHT --repetitions 100 -t -n -p ${CARTDIR}/install/Linux/TESTING" $test
-    run_fg $DAOS_CLIENTS Log/$SLURM_JOB_ID/daos_client_hostlist "${CARTDIR}/install/Linux/TESTING/tests/test_group_np_cli --name client-group --attach_to selftest_srv_grp_$DAOS_SERVERS --shut_only --cfg_path=${CARTDIR}/install/Linux/TESTING" $test
+    run_st $N_CLIENTS Log/$SLURM_JOB_ID/daos_client_hostlist "${CARTDIR}/install/Linux/bin/self_test --group-name selftest_srv_grp_$N_SERVERS --endpoint 0-$last_srv_index:0 --master-endpoint 0:0 --max-inflight-rpcs $MAX_INFLIGHT --repetitions 100 -t -n -p ${CARTDIR}/install/Linux/TESTING" $test
+    run_fg $N_CLIENTS Log/$SLURM_JOB_ID/daos_client_hostlist "${CARTDIR}/install/Linux/TESTING/tests/test_group_np_cli --name client-group --attach_to selftest_srv_grp_$N_SERVERS --shut_only --cfg_path=${CARTDIR}/install/Linux/TESTING" $test
 }
 
 for test in "$@"; do
     print_title $test
     case $test in
         sanity)
             prepare
-            run_fg $DAOS_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "hostname" $test
+            run_fg $N_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "hostname" $test
             ;;
         barrier)
             prepare
-            run_fg $DAOS_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "${CARTDIR}/install/Linux/bin/crt_launch -e ${CARTDIR}/install/Linux/TESTING/tests/test_crt_barrier" $test
+            run_fg $N_SERVERS Log/$SLURM_JOB_ID/daos_server_hostlist "${CARTDIR}/install/Linux/bin/crt_launch -e ${CARTDIR}/install/Linux/TESTING/tests/test_crt_barrier" $test
             ;;
         self_test)
-            let DAOS_SERVER=2
-            until [  $DAOS_SERVERS -gt 65 ]; do
+            let $N_SERVER=2
+            until [  $N_SERVERS -gt 65 ]; do
                 prepare
                 run_selftest
-                let DAOS_SERVERS=$(( DAOS_SERVERS*2 ))
+                let $N_SERVERS=$(( $N_SERVERS*2 ))
             done
 
-            let DAOS_SERVERS=126
+            let $N_SERVERS=126
```

Comment:
My previous comment is suppose to be related to number 126. Are we always expected to run with 126? If we have skx-large what that 126 number will be ? Is that suppose to be N_SERVER -1 ? or am I misunderstanding something related to 126 server?

Reply:
ah... copy and paste from my run for executing on 128 skx-normal cluster.
let me fix this.